### PR TITLE
Feature: align values in column

### PR DESCRIPTION
**Pull request by thorag76**
This change aligns values "column-like" at an index n * indent. 
This improves readability a lot.

### CODE REVIEW

**Comment by geographika** [comment]
@thorag76 - thanks for this. Do you have an example of output before and after this change to highlight the difference?
The CI is currently broken for other reasons.

**Comment by thorag76** [comment]
Hi Seth,

yeah, I was wondering whether it was me who broke the build :D
So thanks for telling me :)

I have attached two files, before|after.map.
The basic idea is that the property with the longest name (in the same level) determines the column to align to - while the columns are the minimum to n * indent possible.

I was thinking about making it configurable - but I decided against it for it looks so much better (to me).
Of course I am open for changes, it's your code :)  ... and I am not 100% sure whether I really covered all cases of "key - value" pairs correctly.

It would be great if you incorporate it but it is okay for me if you don't. I would then go on and keep my fork alive for me and rebase from time to time
[after.map.txt](https://github.com/geographika/mappyfile/files/7997590/after.map.txt)
[before.map.txt](https://github.com/geographika/mappyfile/files/7997592/before.map.txt)
.

File: mappyfile/pprint.py [hunk 1]
```diff
@@ -498,6 +510,22 @@ def _format(self, composite, level=0):
             s = self.whitespace(level, 0) + type_.upper()
             lines.append(s)
 
+        length = 0
```
Comment (geographika) [inline mappyfile/pprint.py:513]: Maybe move this block to its own function for clarity (and would make it easier to test edge cases)

File: mappyfile/pprint.py [hunk 2]
```diff
@@ -199,11 +203,20 @@ def process_dict(self, d, level, comments):
         """
         lines = []
 
+        length = 0
```
Comment (geographika) [inline mappyfile/pprint.py:206]: Add a brief comment on the purpose of this loop

**Comment by geographika** [comment]
Thanks for that example @thorag76 - it makes it much clearer the value of the pull request. 
To get it merged would require the following:

- make it an option on the PrettyPrinter class (set to off by default)
- add at least one test to run with the option on and check basic output

**Comment by thorag76** [comment]
Refactored my code. Found and fixed some nasty bugs as well. 
Provided a basic unit test in test_pprint.py.

**Comment by geographika** [comment]
Thanks @thorag76 !
I'll be switching the CI to GitHub actions at the end of the month, and I'll merge and create a new release with this update then.
